scanner/indexer/indexer.go

```diff
@@ -76,54 +77,79 @@ type Indexer interface {
 // localIndexer is the Indexer implementation that runs libindex locally.
 type localIndexer struct {
 	libIndex        *libindex.Libindex
+	pool            *pgxpool.Pool
+	root            string
 	getLayerTimeout time.Duration
 }
 
 // NewIndexer creates a new indexer.
 func NewIndexer(ctx context.Context, cfg config.IndexerConfig) (Indexer, error) {
+	ctx = zlog.ContextWithValues(ctx, "component", "scanner/backend/indexer.NewIndexer")
+
 	pool, err := postgres.Connect(ctx, cfg.Database.ConnString, "libindex")
 	if err != nil {
 		return nil, fmt.Errorf("connecting to postgres for indexer: %w", err)
 	}
+	defer func() {
+		if err != nil {
```

Comment:
I've seen a pattern in ClairCore where they explicitly set a variable (e.g., `success := false`) for this. I believe it helps avoid confusion on the different scopes where `err` might be defined and ensures the clean-up is explicit. Wanna do that here, too?